src/NewsService.Business/GetNewsCommand.cs

```diff
@@ -15,29 +28,128 @@ public class GetNewsCommand : IGetNewsCommand
+
+        if (response.IsSuccess)
+        {
+          return response.Body.UsersData.FirstOrDefault();
+        }
+
+        _logger.LogWarning(logMessage, string.Join(", ", userId));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, logMessage, string.Join(", ", userId));
+      }
+
+      errors.Add(errorMessage);
+
+      return null;
+    }
+
+    private DepartmentInfo GetDepartment(Guid? departmentId, List<string> errors)
+    {
+      if (departmentId == null)
+      {
+        return null;
+      }
+
+      try
+      {
+        IOperationResult<IGetDepartmentsResponse> departmentResponse =
+          _rcGetDepartments.GetResponse<IOperationResult<IGetDepartmentsResponse>>(
+            IGetDepartmentsRequest.CreateObj(new List<Guid> { departmentId.Value }))
+          .Result.Message;
+
+        if (departmentResponse.IsSuccess)
+        {
+          return _departmentInfoMapper.Map(departmentResponse.Body.Departments.FirstOrDefault());
+        }
+
+        _logger.LogWarning(
+          $"Can not get department. Reason:{Environment.NewLine}{string.Join('\n', departmentResponse.Errors)}.");
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, "Exception on get department request.");
+      }
+
+      errors.Add($"Can not get department info for DepartmentId '{departmentId}'. Please try again later.");
+
+      return null;
+    }
 
     public GetNewsCommand(
       INewsRepository repository,
       INewsResponseMapper mapper,
-      IHttpContextAccessor httpContextAccessor)
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IGetDepartmentsRequest> rcGetDepartments,
+      IRequestClient<IGetUsersDataRequest> rcGetUsers,
+      IDepartmentInfoMapper departmentMapper,
+      ILogger<GetNewsCommand> logger)
     {
       _repository = repository;
       _mapper = mapper;
       _httpContextAccessor = httpContextAccessor;
+      _rcGetDepartments = rcGetDepartments;
+      _rcGetUsers = rcGetUsers;
+      _departmentInfoMapper = departmentMapper;
+      _logger = logger;
     }
 
     public OperationResultResponse<NewsResponse> Execute(Guid newsId)
     {
       OperationResultResponse<NewsResponse> response = new();
 
-      response.Body = _mapper.Map(_repository.Get(newsId));
+      DbNews dbNews = _repository.Get(newsId);
+      if (dbNews == null)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
+        response.Status = OperationResultStatusType.Failed;
+        return response;
+      }
+
+      DepartmentInfo department = null;
+      if (dbNews.DepartmentId.HasValue)
+      {
+        department = GetDepartment(dbNews.DepartmentId, response.Errors);
+      }
+
+      UserData author = GetAuthor(dbNews.AuthorId, response.Errors);
+      if (author == null)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
```

Comment:
what about response.errors?

Reply:
why if we cannot get author we cannot return news?

Reply:
what about response.errors? - done

Reply:
why if we cannot get author we cannot return news? - Because the author cannot be null+